assets/js/modules/analytics-4/utils/withCustomDimensions.js

```diff
@@ -98,15 +98,26 @@ export default function withCustomDimensions( options = {} ) {
 						customDimensions
 					)
 			);
-			const isCreatingCustomDimensions = useSelect(
-				( select ) =>
+			const isAutoCreatingCustomDimensions = useSelect( ( select ) =>
+				select( CORE_FORMS ).getValue(
+					FORM_CUSTOM_DIMENSIONS_CREATE,
+					'isAutoCreatingCustomDimensions'
+				)
+			);
+
+			const isCreatingCustomDimensions = useSelect( ( select ) => {
+				if ( isAutoCreatingCustomDimensions ) {
+					return true;
+				}
```

Comment:
Just to add a bit of separation between the two return possibilities.
```suggestion
				}

```